### PR TITLE
add topic type validation before update topic partitions

### LINKED ISSUE #7152: Increase partitions of a non-existing topic created multiple non-partitioned topics
**Describe the bug**
If I try to increase partitions of a non-existing topic, the method would fail but multiple non-partitioned topics would be created after that.

**To Reproduce**
Assume that we had a standalone Pulsar running. There're no topics at first:

```bash
$ URL_BASE=http://localhost:8080/admin/v2/persistent/public/default
$ curl $URL_BASE; echo
[]
```

Here we use POST method to increase partitions, but the target topic `Foo` doesn't exist. Check topics again, we found there're 3 new topics:

```bash
$ curl $URL_BASE/Foo/partitions -X POST -H 'Content-Type: application/json' -d "3"; echo
{"reason":"Topic is not partitioned topic"}
$ curl $URL_BASE; echo
["persistent://public/default/Foo-partition-0","persistent://public/default/Foo-partition-1","persistent://public/default/Foo-partition-2"]
```

However, it can't be treated as a partitioned topic, so we can't use relative admin APIs to delete it, create a partitioned topic `Foo`, or increase partitions. The only way is to delete each single topic one by one:

```bash
$ curl $URL_BASE/Foo/partitions; echo
{"partitions":0}
$ curl $URL_BASE/Foo/partitions -X DELETE; echo
{"reason":"Partitioned topic does not exist"}
$ curl $URL_BASE/Foo/partitions -X POST -H 'Content-Type: application/json' -d "5"; echo
{"reason":"Already have non partition topicpersistent://public/default/Foo-partition-0 which contains partition suffix '-partition-' and end with numeric value and end with numeric value smaller than the new number of partition. Update of partitioned topic Foo could cause conflict."}
$ curl $URL_BASE/Foo/partitions -X PUT -H 'Content-Type: application/json' -d "5"; echo
{"reason":"This topic already exists"}
$ curl $URL_BASE/Foo-partition-0 -X DELETE
$ curl $URL_BASE/Foo-partition-1 -X DELETE
$ curl $URL_BASE/Foo-partition-2 -X DELETE
$ curl $URL_BASE; echo
[]
```

**Expected behavior**
When we try to increase partitions of a non-existing topic, no topics should be created. The POST method should affect nothing.

### PR DESCRIPTION
**Pull request by aloyszhang**
Fixes #7152 

### Motivation
This PR add topic type validation before update partition numbers of partitioned-topic. 
This can prevent  creating new partitions for non-partitoned topics.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java [hunk 1]
```diff
@@ -252,9 +252,9 @@ public void createNonPartitionedTopic(
      * well. Therefore, it can violate partition ordering at producers until all producers are restarted at application.
      *
      * @param tenant
-     * @param cluster
      * @param namespace
-     * @param topic
+     * @param namespace
```
Comment (vzhikserg) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java:256]: This line is duplicated

Comment (aloyszhang) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java:256]: will fix it

**Pull request merged**